lib/lte_link_control/lte_lc.c

```diff
@@ -235,15 +235,16 @@ static void at_handler_cereg(const char *response)
 	}
 
 	/* Cell update event */
-	if ((cell.id != prev_cell.id) || (cell.tac != prev_cell.tac)) {
+	if ((cell.id != prev_cell.id && cell.id != LTE_LC_CELL_EUTRAN_ID_INVALID) ||
+	    (cell.tac != prev_cell.tac && cell.tac != LTE_LC_CELL_TAC_INVALID)) {
 		evt.type = LTE_LC_EVT_CELL_UPDATE;
 
 		memcpy(&prev_cell, &cell, sizeof(struct lte_lc_cell));
 		memcpy(&evt.cell, &cell, sizeof(struct lte_lc_cell));
 		event_handler_list_dispatch(&evt);
 	}
 
-	if (lte_mode != prev_lte_mode) {
+	if (lte_mode != prev_lte_mode && lte_mode != LTE_LC_LTE_MODE_NONE) {
 		prev_lte_mode = lte_mode;
 		evt.type = LTE_LC_EVT_LTE_MODE_UPDATE;
 		evt.lte_mode = lte_mode;
```

Comment:
The trace has been wrong because `lte_mode` could be `LTE_LC_LTE_MODE_NONE`. So depending on how we leave above condition, this might need to be fixed too.

Reply:
@maxd-nordic, please check that below change to memfault traces is fine.

Reply:
This is fine, Memfault ignores LTE mode changes in the current code